### PR TITLE
implementation of feed and search with token

### PR DESCRIPTION
**Pull request by HamdanSCode**

### CODE REVIEW

File: mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/domain/GetPostsUseCase.kt [hunk 1]
```diff
@@ -3,5 +3,5 @@ package com.hamdan.forzenbook.mainpage.core.domain
 import com.hamdan.forzenbook.core.PostData
 
 interface GetPostsUseCase {
-    suspend operator fun invoke(nameFormat: String): List<PostData>
+    suspend operator fun invoke(nameFormat: String,token:String): List<PostData>
```
Comment (ForzenAcademy) [inline mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/domain/GetPostsUseCase.kt]: FORMAT

File: mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/domain/GetPostsUseCaseImpl.kt [hunk 2]
```diff
@@ -7,7 +7,7 @@ import com.hamdan.forzenbook.mainpage.core.data.repository.FeedRepository
 class GetPostsUseCaseImpl(
     private val repository: FeedRepository
 ) : GetPostsUseCase {
-    override suspend fun invoke(nameFormat: String): List<PostData> {
-        return repository.getFeed(nameFormat).map { it.toPostData() }
+    override suspend fun invoke(nameFormat: String,token:String): List<PostData> {
```
Comment (ForzenAcademy) [inline mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/domain/GetPostsUseCaseImpl.kt]: FORRRRMATTTTT

File: mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/repository/FeedRepositoryImpl.kt [hunk 3]
```diff
@@ -12,12 +12,12 @@ class FeedRepositoryImpl(
     private val feedDao: FeedDao,
     private val userDao: UserDao,
 ) : FeedRepository {
-    override suspend fun getFeed(nameFormat: String): List<Postable> {
+    override suspend fun getFeed(nameFormat: String,token:String): List<Postable> {
```
Comment (ForzenAcademy) [inline mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/repository/FeedRepositoryImpl.kt:15]: RUN LINT GOD DAMNIT

File: searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchViewModel.kt [hunk 4]
```diff
@@ -13,50 +17,94 @@ abstract class BaseSearchViewModel(
...
-            try {
-                searchForPostByIdUseCase(id)
-                onSuccess()
-            } catch (e: Exception) {
-                Log.v("Hamdan", e.message.toString())
-                onError()
-            }
+    private fun searchById(id: Int, context: Context, onSuccess: () -> Unit, onError: () -> Unit) {
+        try {
+            context.getSharedPreferences(
```
Comment (ForzenAcademy) [inline searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchViewModel.kt]: Shared prefs access can block. Put in suspend function please

Comment (HamdanSCode) [reply searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchViewModel.kt]: ok, I didn't realize that, I'll put it in suspend

File: mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/repository/FeedRepositoryImpl.kt (hunk 3 above)
Comment (HamdanSCode) [reply mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/repository/FeedRepositoryImpl.kt:15]: I have been!

File: postCore/src/main/java/com/hamdan/forzenbook/post/core/viewmodel/BasePostViewModel.kt [hunk 5]
```diff
@@ -81,18 +81,21 @@ abstract class BasePostViewModel(
...
-            GlobalConstants.TOKEN_PREFERENCE_LOCATION,
-            Context.MODE_PRIVATE
-        ).getString(TOKEN_KEY, null)
         postState.asImageOrNull()?.let {
             it.filePath?.let { path ->
                 viewModelScope.launch {
                     try {
-                        if (token == null) throw Exception("invalid token")
-                        sendImagePostUseCase(token, path)
+                        context.getSharedPreferences(
```
Comment (ForzenAcademy) [inline postCore/src/main/java/com/hamdan/forzenbook/post/core/viewmodel/BasePostViewModel.kt:88]: Coroutine?

**Review by ForzenAcademy — APPROVED** [review]

**Pull request merged**